docs/topics/enum-classes.md

```diff
@@ -103,9 +115,41 @@ inline fun <reified T : Enum<T>> printAllValues() {
 printAllValues<RGB>() // prints RED, GREEN, BLUE
 ```
 
-Every enum constant has properties for obtaining its name and position (starting with 0) in the enum class declaration:
+If you don't want your enum constants returned in an array, you can use the `entries` property instead. The
+`entries` property returns a pre-allocated immutable list of your enum constants. This is particularly useful when you
+are working with [collections](collections-overview.md) and can you help you avoid [performance issues](https://github.com/Kotlin/KEEP/blob/master/proposals/enum-entries.md#examples-of-performance-issues).
 
+For example:
 ```kotlin
-val name: String
-val ordinal: Int
-```
\ No newline at end of file
+enum class RGB { RED, GREEN, BLUE }
+
+@OptIn(ExperimentalStdlibApi::class)
+fun main() {
+    for (color in RGB.entries) println(color.toString())
+    // prints RED, GREEN, BLUE
+}
+```
+
+> The `entries` property is [Experimental](components-stability.md#stability-levels-explained). To use it you need 
```

Comment:
Hmm, there is a problem that this compatibility guide is a bit incorrect and confuses technologies such as Kotlin/JVM and certain language features which should have their own definitions of compatibility levels. We are thinking about how to redesign it to include correct information about language features. I'd propose leaving the text as is, but remove the link to this guide, there should be enough explanation in `ExperimentalStdlibApi` annotation